CLOUD-215 Add IO functions to read and write text of type string

ibadami opened the pull request:
- Converts all the three functions Dump, readHeader and ReadModel to DumpText, readHeaderText and ReadModelFromText

ibadami committed:
CLOUD-215 Add io functions to read and write text of type string

- Converts all the three functions Dump, readHeader and ReadModel to DumpText, readHeaderText and ReadModelFromText

stelioskat commented:
I think that it is more fitting to use a bit different naming scheme for the new functions. Instead of using the word "Text" use "String" E.g. `NewModelFromText` might be better to be called `NewModelFromString`.

Furthermore, it would be nice to also add documentation to all fun exported elements, at least for files `model.go` and `modelio.go` to make a start :)

sertopexamgio commented on modelio.go:
```diff
@@ -340,3 +340,301 @@ func (model *Model) ReadModel(file string) error {
+
+	var l int = model.l
+	output = append(output, fmt.Sprintf("total_sv %d\n", l))
+
```
Don't we need `if len(model.rho) > 0 {` here ?

sertopexamgio commented on modelio.go:
Do we need this line? Because we have it again on the next line.

sertopexamgio commented on modelio.go:
typo: `match`

sertopexamgio commented on modelio.go:
Thus we use `line` only here, a small optimization could be to use here text[l] and erase the variable line from the `for statement`.

ibadami commented:
Great reviews guys, thanks!

ibadami committed:
Minor fixes and documentation for new function

- Changed name of the functions using string instead of text.
- Added documentation in the new and related old functions.

ibadami committed:
Check for len(model.rho)

ibadami committed:
Remove redundant code and if checks

ibadami committed:
Change import from original to forked

ibadami committed:
Add if checks for each field to avoid dumping the empty field names

ibadami committed:
Check len of SvCoef

ibadami commented:
No, SV should be there. If there are no Support vectors after the SV entry, the readingFunction will throw error. But reading function must read until SV.
